targets/psa_switch/psa_switch.cpp

```diff
@@ -141,23 +144,16 @@ PsaSwitch::receive_(port_t port_num, const char *buffer, int len) {
                                bm::PacketBuffer(len + 512, buffer, len));
 
   BMELOG(packet_in, *packet);
-
   PHV *phv = packet->get_phv();
-  // many current P4 programs assume this
-  // it is also part of the original P4 spec
   phv->reset_metadata();
```

Comment:
I think if you remove the comment, you should also remove the call to `reset_metadata`. Plus, according to the PSA spec:

> The PSA architecture does not mandate initialization of user-defined metadata to known values as given as input to the ingress parser.

Reply:
Maybe @jafingerhut can comment on this

Reply:
I updated this comment to include this excerpt, originally (and eagerly) removed the comment thinking that it's implied

Reply:
So the PSA architecture does not require initialization of user-defined metadata.

A particular PSA _implementation_ still can do so, if the implementer wants to do that.

Some tradeoffs:

If all user-defined metadata is initialized to a well known value, it makes the P4 programs more predictable in their behavior.  However, if they want to catch portability bugs in their programs where they are using uninitialized values, such an implementation will not help them much.  However, even assigning some kind of "random" values intentionally as initial values for user-defined metadata is not guaranteed to catch such bugs -- you really want a static analysis tool for that purpose, which do exist (e.g. p4v, p4pktgen).

Conclusion:

I think it is reasonable for psa_switch to initialize all user-defined metadata to fixed values, e.g. 0 for `bit<W>` and `int<W>` types, and whatever the bit pattern 0 in psa_switch memory means for `enum` types (maybe no legal value, which is OK for an implementation to do).  It would be good to have a comment that while this is not required behavior for a PSA implementation, it is allowed behavior, and someone may want to consider changing that behavior in psa_switch later.